sqrl-planner/sqrl-planner-local/src/test/resources/snapshots/com/datasqrl/plan/local/analyze/QuerySnapshotTest/customerRecentProductsTest.txt

```diff
@@ -10,19 +10,22 @@ LogicalTableScan(table=[[product$1]])
-          LogicalJoin(condition=[=($12, $0)], joinType=[inner])
+    LogicalProject(__pk1$_uuid=[$17], productid=[$13], category=[COALESCE($37, '')], quantity=[$14], _uuid=[$0], _idx=[$12], _uuid1=[$32], __timestamp38=[CASE(<($31, $33), $33, $31)])
+      LogicalJoin(condition=[=($13, $34)], joinType=[left])
+        LogicalProject(_uuid=[$0], _ingest_time=[$1], id=[$2], customerid=[$3], time=[$4], entries=[$5], _uuid0=[$6], _ingest_time0=[$7], id0=[$8], customerid0=[$9], time0=[$10], entries0=[$11], _idx=[$12], productid=[$13], quantity=[$14], unit_price=[$15], discount=[$16], _uuid00=[$17], _ingest_time00=[$18], customerid00=[$19], email=[$20], name=[$21], lastUpdated=[$22], _uuid000=[$23], _ingest_time000=[$24], id00=[$25], customerid000=[$26], time00=[$27], entries00=[$28], __timestamp=[$29], __timestamp24=[$30], __timestamp31=[CASE(<($4, $30), $30, $4)])
+          LogicalJoin(condition=[=($6, $0)], joinType=[inner])
             LogicalFilter(condition=[<(NOW(), +($4, 31536000000:INTERVAL SECOND(11)))])
               LogicalTableScan(table=[[orders$2]])
-            LogicalProject(_uuid=[$0], _ingest_time=[$1], customerid=[$2], email=[$3], name=[$4], lastUpdated=[$5], _uuid0=[$6], _ingest_time0=[$7], id=[$8], customerid0=[$9], time=[$10], entries=[$11], _idx=[$12], productid=[$13], quantity=[$14], unit_price=[$15], discount=[$16], __timestamp=[CASE(<($1, $10), $10, $1)])
-              LogicalJoin(condition=[=($2, $9)], joinType=[inner])
-                LogicalTableScan(table=[[customer$2]])
-                LogicalCorrelate(correlation=[$cor7], joinType=[inner], requiredColumns=[{5}])
+            LogicalProject(_uuid=[$0], _ingest_time=[$1], id=[$2], customerid=[$3], time=[$4], entries=[$5], _idx=[$6], productid=[$7], quantity=[$8], unit_price=[$9], discount=[$10], _uuid0=[$11], _ingest_time0=[$12], customerid0=[$13], email=[$14], name=[$15], lastUpdated=[$16], _uuid00=[$17], _ingest_time00=[$18], id0=[$19], customerid00=[$20], time0=[$21], entries0=[$22], __timestamp=[$23], __timestamp24=[CASE(<($4, $23), $23, $4)])
+              LogicalJoin(condition=[=($17, $0)], joinType=[inner])
+                LogicalCorrelate(correlation=[$cor5], joinType=[inner], requiredColumns=[{5}])
                   LogicalTableScan(table=[[orders$2]])
                   Uncollect
-                    LogicalProject($f0=[$cor7.entries])
+                    LogicalProject($f0=[$cor5.entries])
                       LogicalValues(tuples=[[{ 0 }]])
+                LogicalProject(_uuid=[$0], _ingest_time=[$1], customerid=[$2], email=[$3], name=[$4], lastUpdated=[$5], _uuid0=[$6], _ingest_time0=[$7], id=[$8], customerid0=[$9], time=[$10], entries=[$11], __timestamp=[CASE(<($1, $10), $10, $1)])
+                  LogicalJoin(condition=[=($2, $9)], joinType=[inner])
```

Comment:
Looks like there is an extra join in here, right?

Reply:
It's correct, I was adding in another table that was incorrectly scoped before but it just so happened to be eliminated better.